__tests__/AdminLogin.tsx

```diff
@@ -0,0 +1,81 @@
+import { screen } from '@testing-library/react';
+import { MapiAuthConnectors } from 'lib/MapiAuth/MapiAuth';
+import TestOAuth2 from 'lib/OAuth2/TestOAuth2';
+import { getInstallationInfo } from 'model/services/giantSwarm/info';
+import { getConfiguration } from 'model/services/metadata/configuration';
+import { MainRoutes } from 'shared/constants/routes';
+import {
+  AWSInfoResponse,
+  getMockCall,
+  metadataResponse,
+  ORGANIZATION,
+  orgResponse,
+  orgsResponse,
+  releasesResponse,
+} from 'testUtils/mockHttpCalls';
+import {
+  createInitialHistory,
+  renderRouteWithStore,
+} from 'testUtils/renderUtils';
+
+describe('AdminLogin', () => {
+  it('performs the admin login flow via OAuth2', async () => {
+    (getInstallationInfo as jest.Mock).mockResolvedValueOnce(AWSInfoResponse);
+    (getConfiguration as jest.Mock).mockResolvedValueOnce(metadataResponse);
+    getMockCall('/v4/organizations/', orgsResponse);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getMockCall(`/v4/organizations/${ORGANIZATION}/`, orgResponse as any);
+    getMockCall(`/v4/organizations/${ORGANIZATION}/credentials/`);
+    getMockCall('/v4/clusters/');
+    getMockCall('/v4/appcatalogs/');
+    getMockCall('/v4/releases/', releasesResponse);
+
+    const history = createInitialHistory(MainRoutes.AdminLogin);
+
+    const testAuth = new TestOAuth2(history);
+    const attemptLoginMockFn = jest.spyOn(testAuth, 'attemptLogin');
+
+    renderRouteWithStore(
+      MainRoutes.AdminLogin,
+      undefined,
+      undefined,
+      testAuth,
+      history
+    );
+
+    expect(attemptLoginMockFn).toHaveBeenCalledWith(
```

Comment:
Since we are using a different OAuth2 provider for testing (`TestOAuth2`, instead of `MapiAuth`, to not have to test the `oidc-client-js` library), there's no way to tell if we are using the admin login (which is executed by adding an extra parameter in the URL), without actually replicating the logic in `TestOAuth2`.

If we would actually replicate the `MapiAuth` logic in `TestOAuth2` and test the "behaviour" - if the browser redirects to the correct URL, then we would just be testing the test.

By checking if the correct connector is used, we know that we call `MapiAuth` correctly. It's its job to do the right thing starting then.